### ISSUE TITLE
tags & categories not working in github pages

### ISSUE THREAD

**Issue by deathflash1411** [cotes2020/jekyll-theme-chirpy#43]
Hey,

after uploading to github pages the tags & categories are not working.
both return a "404: Page not found" error

**Comment by bstankie** [comment]
I'm getting the same error. It seems that for the Categories page it is inserting 'localhost:4000' in the href link (see below). 
<a href="http://localhost:4000/posts/Academia-Industry/">Academia vs. Industry</a>

You can see the behavior here: http://www.bstankie.com/categories/career/

Tags are also having issues that I haven't been able to diagnose. 

I've read through the manual and couldn't find any information other than: https://chirpy.cotes.info/posts/write-a-new-post/

I've configured those pages as described.

**Comment by cotes2020** [comment]
@bstankie There's a mistake in docs <https://chirpy.cotes.info/posts/write-a-new-post/>:

> the content of a tag type page is:
> ```
> ---
> layout: tag
> title: TAG_NAME
> category: TAG_NAME
> ---
> ```

The correct way is to change the `category` above to `tag`, my bad; I'll fix the docs online.

**Comment by bstankie** [comment]
OK. I'm still having trouble. Here are the steps that I'm going through. See https://bstankie.com

Here are the steps, output and the behavior.

1. Create a post with a category and a tag.

```yaml
---
layout: post
title:  "Academia vs. Industry"
date:   2019-08-01 12:21:24 -0600
categories: [career]
tags: [academia]
---
```
see (webpage)[http://www.bstankie.com/posts/Academia-Industry/]

2. Create a file `academia.html` in the `tags` directory with the following information:

## academia.html
```yaml
---
layout: tag
title: academia             # e.g. bee
tag: academia               # e.g. bee
---
```

3. Create a file `career.html` in the `categories` directory with the following information:

## career.html
```yaml
---
layout: category
title: career        # e.g. Insect
category: career     # e.g. Insect
---
```

4. run `tools/init.sh`

```bash    
bstankie@DESKTOP-134K253:/mnt/d/git/BJS-Blog-chirpy$ tools/init.sh
[INFO] Succeed! 0 category-pages created.
[INFO] Succeed! 0 tag-pages created.
[master 6c7f058] [Automation] Updated the Categories, Tags for post(s).
 2 files changed, 10 deletions(-)
 delete mode 100644 categories/career.html
 delete mode 100644 tags/academia.html
Updated the Categories, Tags for post(s).
```

It seems to say `Succeed!` but there are no category nor tag pages created.

5. run `tools/build.sh`

```bash
bstankie@DESKTOP-134K253:/mnt/d/git/BJS-Blog-chirpy$ tools/build.sh
The dependency tzinfo-data (>= 0) will be unused by any of the platforms Bundler is installing for. Bundler is installing for ruby but the dependency is only for x86-mingw32, x86-mswin32, x64-mingw32, java. To add those platforms to the bundle, run `bundle lock --add-platform x86-mingw32 x86-mswin32 x64-mingw32 java`.
Configuration file: /mnt/d/git/BJS-Blog-chirpy/_config.yml
           Cleaner: Removing /mnt/d/git/BJS-Blog-chirpy/_site...
           Cleaner: Nothing to do for /mnt/d/git/BJS-Blog-chirpy/.jekyll-metadata.
           Cleaner: Nothing to do for /mnt/d/git/BJS-Blog-chirpy/.jekyll-cache.
           Cleaner: Nothing to do for .sass-cache.
$ cd /mnt/d/git/BJS-Blog-chirpy/.container
$ JEKYLL_ENV=production bundle exec jekyll b -d /mnt/d/git/BJS-Blog-chirpy/_site
Configuration file: /mnt/d/git/BJS-Blog-chirpy/.container/_config.yml
            Source: /mnt/d/git/BJS-Blog-chirpy/.container
       Destination: /mnt/d/git/BJS-Blog-chirpy/_site
 Incremental build: disabled. Enable with --incremental
      Generating...
                    done in 1.255 seconds.
 Auto-regeneration: disabled. Use --watch to enable.

Build success, the site files have been placed in '/mnt/d/git/BJS-Blog-chirpy/_site'.
```

6. Upload to AWS S3
7. See behavior:
    1. [bstankie.com](http://www.bstankie.com/)
    2. [bstankie tags](http://www.bstankie.com/tabs/tags/): Click on `academia` tag and page not found
    3. [bstankie categories](http://www.bstankie.com/tabs/categories/): Click on `career` tag and page not found

**Comment by cotes2020** [comment]
@bstankie  it seems that you don't understand what the docs saying.

1. The script `tools/init.sh`  is used to automatically create pages we need so that you can completely save the time of manual creation.

2. I just check the post source file (<https://raw.githubusercontent.com/bstankie/BJS-Blog-chirpy/master/_posts/2019-08-01-Academia-Industry.markdown>) of your repo and found the cause of the problems:
- the extend of post file name should be `.md`.
- the value  list of `categoires` in the fromatter block should be arounded by `[` and `]`
  ```
  ----
  categoires: [career]   # or category: career
  ---
  ```

**Comment by bstankie** [comment]
The .md seemed to be the main issue. Tried with and without the `[ ]` before and ran into the same issues. 

Now running into a strange problem. The URL when linking from the [category:career](http://www.bstankie.com/categories/career/) to the post is: 

http://www.bstankie.com/categories/career/www.bstankie.com/posts/Academia-Industry/

The [tags:academia](http://www.bstankie.com/tags/academia/) has a similar issue of linking to:

http://www.bstankie.com/tags/academia/www.bstankie.com/posts/Academia-Industry/

I do appreciate your help on this. Wish I were able to figure it out

**Comment by cotes2020** [comment]
Check the variable `url` of `_config.yml`, I sure that you're missing the protocol part 'https://'

**Comment by bstankie** [comment]
That did it.

Thanks @cotes2020 ! Would you like me to document any of this in the readme to pay it forward?

**Comment by cotes2020** [comment]
Of course! I can't wait for somebody join in the doc improvement stuff. 😄

**Comment by cotes2020** [comment]
A tip for your website:  [Enforcing HTTPS for your GitHub Pages site](https://help.github.com/en/github/working-with-github-pages/securing-your-github-pages-site-with-https#enforcing-https-for-your-github-pages-site)

**Comment by bstankie** [comment]
Thanks for the HTTPS tip. Working on that next. Right now it is being deployed through AWS.

[Coffee](https://www.buymeacoffee.com/coteschung) sent as a small token of gratitude.

**Comment by cotes2020** [comment]
I got the coffee, thanks !

**Comment by Cloufish** [comment]
I'm having the same issue of having 404 in categories. I've got the proper format of categories and tags, ran tools/init.sh script. But there's still a 404 error in GitHub Pages. Could you take a look cotes in my blog?

https://cloufish.github.io/tabs/categories/

**Comment by Onurbolatogluu** [comment]
@Cloufish I am having the same problem. how did you solve the problem can you help

**Comment by Cloufish** [comment]
From your _config.yml file I can see that you haven't even filled it! :P. I'm my case it was because I've filled baseurl: parameter which in my makes shouldn't be filled.

**Comment by Onurbolatogluu** [comment]
@Cloufish Thanks my friend for your help, I couldn't find the problem but I copied your project :)

**Comment by Onurbolatogluu** [comment]
Brother, I think I can't see it, can you help me change what should I change? :/

**Comment by Cloufish** [comment]
I don't think copying is the best solution, because you will eventually come into another issue :p However, if it's less complicated for you then go-ahead

**Comment by Cloufish** [comment]
See the 'url:' parameter inside _config.yml? Change it to ' https://onurbolatogluu.github.io ' and you'll be ready to go :100:

**Comment by Onurbolatogluu** [comment]
@Cloufish Brother, I changed it as you said, I am sending it in the attachment. The only difference is I use my own domain name. I organized it accordingly. but the problem still persists. May I be very unlucky :)
![https](https://user-images.githubusercontent.com/46894029/95001466-ea78fa00-05d2-11eb-8c23-1caaaea2c58e.PNG)
![site](https://user-images.githubusercontent.com/46894029/95001471-02507e00-05d3-11eb-93d3-7fddc451be77.PNG)
